--- binutils-2.9.1/gas/config/tc-m68k.c.gfl	Fri May  1 17:44:42 1998
+++ binutils-2.9.1/gas/config/tc-m68k.c	Sat Feb 19 17:55:46 2000
@@ -3129,14 +3129,18 @@
     }
 #endif
 
-  symbol_table_insert (symbol_new (regname, reg_section, regnum,
+  /* Use symbol_create here instead of symbol_new so we don't try to
+     output registers into the object file's symbol table.  */
+  symbol_table_insert (symbol_create (regname, reg_section, regnum,
 				   &zero_address_frag));
 
   for (i = 0; regname[i]; i++)
     buf[i] = islower (regname[i]) ? toupper (regname[i]) : regname[i];
   buf[i] = '\0';
 
-  symbol_table_insert (symbol_new (buf, reg_section, regnum,
+  /* Use symbol_create here instead of symbol_new so we don't try to
+     output registers into the object file's symbol table.  */
+  symbol_table_insert (symbol_create (buf, reg_section, regnum,
 				   &zero_address_frag));
 }
 
--- binutils-2.9.1/gas/config/obj-aout.c.gfl	Fri May  1 17:44:34 1998
+++ binutils-2.9.1/gas/config/obj-aout.c	Sat Feb 19 18:31:18 2000
@@ -487,7 +487,12 @@
 	 All other symbols are output.  We complain if a deleted
 	 symbol was marked external. */
 
-
+      /* Unfortunately the definition of N_REGISTER in struc-symbol.h
+         conflicts with N_WARNING in gnu_aout.h.  But if register 
+         symbols have been properly created with symbol_create instead
+         of symbol_new we don't  have to check for register symbols
+         again here.  */
+#if 0
       if (!S_IS_REGISTER (symbolP)
 	  && (!S_GET_NAME (symbolP)
 	      || S_IS_DEBUG (symbolP)
@@ -495,6 +500,14 @@
 	      || S_IS_EXTERNAL (symbolP)
 	      || (S_GET_NAME (symbolP)[0] != '\001'
 		  && (flag_keep_locals || !S_LOCAL_NAME (symbolP)))))
+#else
+      if (!S_GET_NAME (symbolP)
+	      || S_IS_DEBUG (symbolP)
+	      || !S_IS_DEFINED (symbolP)
+	      || S_IS_EXTERNAL (symbolP)
+	      || (S_GET_NAME (symbolP)[0] != '\001'
+		  && (flag_keep_locals || !S_LOCAL_NAME (symbolP))))
+#endif
 	{
 	  symbolP->sy_number = symbol_number++;
 
